src/Intercom/Provider.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace SocialiteProviders\Intercom;
+
+use SocialiteProviders\Manager\OAuth2\User;
+use Laravel\Socialite\Two\ProviderInterface;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+
+class Provider extends AbstractProvider implements ProviderInterface
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'INTERCOM';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [''];
```

Comment:
The `$scopes` array can be removed as it is just an empty array like already defined here https://github.com/laravel/socialite/blob/3.0/src/Two/AbstractProvider.php#L62.